src/engine.rs

```diff
@@ -907,6 +907,112 @@ impl Engine {
         serde_json::to_string_pretty(&ast).map_err(anyhow::Error::msg)
     }
 
+    /// Get the package names of each policy added to the engine.
+    ///
+    ///
+    /// ```rust
+    /// # use regorus::*;
+    /// # use anyhow::{bail, Result};
+    /// # fn main() -> Result<()> {
+    /// # let mut engine = Engine::new();
+    /// engine.add_policy("test.rego".to_string(), "package test\n x := 1".to_string())?;
+    /// engine.add_policy("test2.rego".to_string(), "package test.multi.segment\n x := 1".to_string())?;
+    ///
+    /// let package_names = engine.get_policy_package_names()?;
+    /// let value = Value::from_json_str(&package_names)?;
+    ///
+    /// assert_eq!(value[0]["package_name"].as_string()?.as_ref(), "test");
+    /// assert_eq!(value[1]["package_name"].as_string()?.as_ref(), "test.multi.segment");
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[cfg(feature = "ast")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "ast")))]
```

Comment:
This functionality does not need to be guarded with the `ast` feature.

However, if this is experimental, we can introduce a new feature to guard it.